### ISSUE TITLE
TabView: wrong activeIndex on tab close event

### ISSUE THREAD

**Issue by crocimi** [primefaces/primefaces#2161]
## 1) Environment
- PrimeFaces version:primefaces-6.0.16
- Does it work on the newest released PrimeFaces version? No
- Does it work on the newest sources in GitHub? (Build by source -> https://github.com/primefaces/primefaces/wiki/Building-From-Source)
- Application server + version: Wildfly 10.0.0
- Affected browsers: all

## 2) Expected behavior
wrong activeIndex if a non-selected tab is removed from the tabView

## 3) Actual behavior
activeIndex should be decremented by one 

## 4) Steps to reproduce
Select Tab with index=4
Close Tab with index 0 (with direct "click" on the close button)
.. 

## 5) Sample XHTML

 <h:form>

          <p:tabView value="#{tabbedView.cars}" var="item" scrollable="true" activeIndex="#{tabbedView.activeIndex}" dynamic="true">
              <p:ajax event="tabChange" listener="#{tabbedView.onTabChange}"/>
              <p:ajax event="tabClose"  listener="#{tabbedView.onTabClose}"/>

              <p:tab title="#{item}" closable="true">

              </p:tab>
          </p:tabView>

## 6) Sample bean
@ManagedBean
@ViewScoped
public class TabbedView {

    private int activeIndex;

    private List<String> cars;

    @PostConstruct
    public void init() {
        cars = new ArrayList<String>();
        cars.add("Fiat");
        cars.add("BMW");
        cars.add("Ford");
        cars.add("Volvo");
        cars.add("Jaguar");
    }

    public void onTabChange(TabChangeEvent event) {
        FacesMessage msg = new FacesMessage("Tab Changed", "Active Tab: " + event.getTab().getTitle());
        FacesContext.getCurrentInstance().addMessage(null, msg);
    }

    public void onTabClose(TabCloseEvent event) {
        FacesMessage msg = new FacesMessage("Tab Closed", "Closed tab: " + event.getTab().getTitle());
        FacesContext.getCurrentInstance().addMessage(null, msg);
    }

    public int getActiveIndex() {
        return activeIndex;
    }

    public void setActiveIndex(int activeIndex) {
        this.activeIndex = activeIndex;
    }

    public List<String> getCars() {
        return cars;
    }

    public void setCars(List<String> cars) {
        this.cars = cars;
    }

[TabbedView.java.txt](https://github.com/primefaces/primefaces/files/833911/TabbedView.java.txt)
[tabView.xhtml.txt](https://github.com/primefaces/primefaces/files/833912/tabView.xhtml.txt)

**Comment by tandraschko** [comment]
Would be great if you could provide a PR

**Comment by tandraschko** [comment]
Works fine for me in trunk.

**Comment by ggeremy** [comment]
Hello,

I think there is a related problem on Primefaces Showcase.
I found a NullPointerException on the "Events" example from TabView demo.

Steps to reproduce:
 * close the "Godfather Part II" -> tab is closed
 * close the "Godfather Part III" -> tab is not closes
 * try to close again "Godfather Part III" -> NPE 


By the way, I've also experienced a problem with the TabCloseEvent on my application.
On server side, I log the tab title on TabCloseEvent but it doesn't match with the tab title I am actually closing.
